### PR TITLE
+ getAll() settings method

### PR DESCRIPTION
**Pull request by JFixby**
Reads all values from cache.

### CODE REVIEW

**Comment by JFixby** [comment]
I signed it!

**Comment by JFixby** [comment]
I signed it!

File: packages/shared_preferences/lib/shared_preferences.dart [hunk 1]
```diff
@@ -44,6 +44,9 @@ class SharedPreferences {
   /// in sync since the setter method might fail for any reason.
   final Map<String, Object> _preferenceCache;
 
+  /// Reads all values from cache
+  Map<String, Object> getAll() => new Map.from(_preferenceCache);
```
Comment (mit-mit) [inline packages/shared_preferences/lib/shared_preferences.dart]: This is causing an analysis error (see the travis log, or run `flutter analyze` from your sources). You are missing a type annotation on the right-hand side; see the `getAll()` method for an example.

Comment (JFixby) [inline packages/shared_preferences/lib/shared_preferences.dart]: I already tried:

```Dart
Map<String, Object> result = new Map<String, Object>();
result.addAll(_preferenceCache);
```

```Dart
final result = <String, Object>{};
result.addAll(_preferenceCache);
```

```Dart
final Map<String, Object> result = {};
result.addAll(_preferenceCache);
```

and the 

```Dart
Map<String, Object> getAll() => new Map.from(_preferenceCache);
```

I'm out of ideas.

**Comment by JFixby** [comment]
I already tried:

```Dart
Map<String, Object> result = new Map<String, Object>();
result.addAll(_preferenceCache);
```

```Dart
final result = <String, Object>{};
result.addAll(_preferenceCache);
```

```Dart
final Map<String, Object> result = {};
result.addAll(_preferenceCache);
```

and the 

```Dart
Map<String, Object> getAll() => new Map.from(_preferenceCache);
```

I'm out of ideas.

**Comment by mit-mit** [comment]
I believe the right solution is `Map<String, Object> getAll() => new Map<String, Object>.from(_preferenceCache);`

@floitschG can you confirm?

**Comment by mit-mit** [comment]
@JFixby I got confirmation that `Map<String, Object> getAll() => new Map<String, Object>.from(_preferenceCache);` is the right fix.

**Comment by mit-mit** [comment]
Also, we hope to some day have the smarts to infer the type so that this shouldn't be needed, but right now it is.

**Comment by mit-mit** [comment]
Sorry, didn't mean to close

**Comment by JFixby** [comment]
That one also failed.

**Comment by mit-mit** [comment]
No, the analysis job passed! But looks like there might be a formatting issue, see the Travis log.

**Pull request closed without merging**